ldap/servers/slapd/back-ldbm/ldbm_modrdn.c

```diff
@@ -1557,6 +1566,204 @@ ldbm_back_modrdn(Slapi_PBlock *pb)
     return retval;
 }
 
+
+/* dsEntryDN functions */
+
+static int
+dsentrydn_moddn_rename_child(
+    back_txn *ptxn,
+    Slapi_Backend *be,
+    struct ldbminfo *li,
+    struct backentry *e,
+    struct backentry **ec,
+    int parentdncomps,
+    char **newsuperiordns,
+    int newsuperiordncomps)
+{
+    /*
+     * Construct the new DN for the entry by taking the old DN
+     * excluding the old parent entry DN, and adding the new
+     * superior entry DN.
+     */
+    int retval = 0;
+    char *olddn;
+    char *newdn;
+    char **olddns;
+    int olddncomps = 0;
+    int need = 1; /* For the '\0' */
+    size_t i;
+
+    olddn = slapi_entry_attr_get_charptr((*ec)->ep_entry, SLAPI_ATTR_DS_ENTRYDN);
+    if (NULL == olddn) {
+        return retval;
+    }
+    olddns = slapi_ldap_explode_dn(olddn, 0);
+    if (NULL == olddns) {
+        goto out;
+    }
+    for (; olddns[olddncomps] != NULL; olddncomps++);
+
+    for (i = 0; i < olddncomps - parentdncomps; i++) {
+        need += strlen(olddns[i]) + 1; /* For the "," */
+    }
+    for (i = 0; i < newsuperiordncomps; i++) {
```

Comment:
any reason not to scope the size_t i = 0; here per loop rather than once outside? IIRC there is some compiler optimisations that can occur if you scope it per loop rather than once and try to reuse it.